kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -452,6 +513,16 @@ class KinematicsBase
   }
 
   /**
+   * @brief Sets the discretization applied to each redundant joint
+   * @param discretization a map of joint indices and discretization value pairs.  The first pair element corresponds
+   * to the index of the joint to be discretized.
```

Comment:
We have 7DOF robots attached to linear rails.  It's this exact use case that requires different discretization.  A linear rail might be 0-10m, and a revolute jiont -Pi - PI rads.  The difference in units and/or scale necessitates a different discretization.